consumer/pdata/metric.go

```diff
@@ -35,10 +37,114 @@ func (at AggregationTemporality) String() string {
 //
 // Outside of the core repository the metrics pipeline cannot be converted to the new model since data.MetricData is
 // part of the internal package.
-//
-// IMPORTANT: Do not try to convert to/from this manually, use the helper functions in the pdatautil instead.
 type Metrics struct {
-	InternalOpaque interface{}
+	orig *[]*otlpmetrics.ResourceMetrics
+}
+
+// MetricDataFromOtlp creates the internal MetricData representation from the OTLP.
+func MetricsFromOtlp(orig []*otlpmetrics.ResourceMetrics) Metrics {
+	return Metrics{&orig}
+}
+
+// MetricDataToOtlp converts the internal MetricData to the OTLP.
+func MetricsToOtlp(md Metrics) []*otlpmetrics.ResourceMetrics {
+	return *md.orig
+}
```

Comment:
Can we use the same approach as this https://github.com/open-telemetry/opentelemetry-collector/pull/1703 to make this functions limited for use in this repo only?

Reply:
Will do in a separate PR.